File: java/app/src/main/java/org/example/beginner/Q01.java
package org.example.beginner;

import java.util.Objects;

public class Q01 {
  public static void main(String[] args) {
    var begin = System.currentTimeMillis();

    for (int i = 10;; i++) {
      var b02 = Integer.toString(i, 2);
      var o08 = Integer.toString(i, 8);
      var d10 = Integer.toString(i, 10);
      var x16 = Integer.toString(i, 16);

      if (Objects.equals(b02, new StringBuilder(b02).reverse().toString())
          && Objects.equals(o08, new StringBuilder(o08).reverse().toString())
          && Objects.equals(d10, new StringBuilder(d10).reverse().toString())) {
        System.out.println("a01. = " + i + ", " + x16);
        break;
      }
    }

    var finish = System.currentTimeMillis();
    System.out.println((finish - begin) + "ms");
  }
}

File: script/bun/src/puzzle/beginner/q01.ts
/**
 * 10進数、8進数、2進数、すべてで回文数となる数値のうち、
 * 10進数の10以上で最小の数は？
 */

const begin = performance.now();

for (let i = 10; ; i++) {
  const b02 = i.toString(2);
  const o08 = i.toString(8);
  const d10 = i.toString(10);
  const x16 = i.toString(16);

  if (
    b02 === b02.split('').reverse().join('') &&
    o08 === o08.split('').reverse().join('') &&
    d10 === d10.split('').reverse().join('')
  ) {
    console.log(`a01. = ${i}, ${x16}`);
    break;
  }
}

const finish = performance.now();
console.log(`${finish - begin}ms`);
